app-cpp/app355.h

```diff
@@ -6,6 +6,123 @@
 #define _snprintf_s (snprintf)
 #endif // WIN32
 
+// std
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <list>
+#include <mutex>
+#include <string>
+#include <tuple>
+#include <thread>
+#include <vector>
+
+// self
+#include "libsat355.h"
+
+#pragma region class OrbitalData
+class OrbitalData
+{
+private:
+    sat355::TLE mTLE;
+    double mLatitude;
+    double mLongitude;
+    double mAltitude;
+    std::string mName;
+    double mMeanMotion;
+
+public:
+    OrbitalData(sat355::TLE inTLE, double latitude, double longitude, double altitude)
+        : mTLE{std::move(inTLE)}, mLatitude(latitude), mLongitude(longitude), mAltitude(altitude)
+    {
+        mName = mTLE.GetName();
+        mMeanMotion = mTLE.GetMeanMotion();
+    }
+
+    OrbitalData() = delete;
+
+    ~OrbitalData() = default;
+
+    const sat355::TLE& GetTLE() const 
+    { 
+        return mTLE; 
+    }
+
+    double GetLatitude() const 
+    { 
+        return mLatitude; 
+    }
+    void SetLatitude(double latitude) 
+    { 
+        mLatitude = latitude; 
+    }
+
+    double GetLongitude() const 
+    { 
+        return mLongitude; 
+    }
+    void SetLongitude(double longitude) 
+    { 
+        mLongitude = longitude; 
+    }
+
+    double GetAltitude() const 
+    { 
+        return mAltitude; 
+    }
+    void SetAltitude(double altitude) 
+    { 
+        mAltitude = altitude; 
+    }
+};
+#pragma endregion {}
+
+namespace {
+//--------------------------------------------------
```

Comment:
this is a terrible bug!
anonymous namespace.... used in a header file, is a terrible bug!
anonymous namespace should only be used within a `.cpp`!

basically, when writing c++... all your c++ code should be in a namespace of some sort.
(except for `main()`, which is the only thing that should be at naked global ("C") namespace)

- your **interfaces** will be described in a `.hpp` using 1) some interface-specific namespace with a name of your choosing.
- your **implementations** will be described in a `.cpp`, using both the 1) interface-specifc namespace _and_ 2) anonymous namespaces.

- Functions _declared_ in the interface will be _defined_ in the `.cpp` inside the 1) interface-specific namespace.
- Functions or classes that are pure implementation (that have no visibility in the interface) are _declared+defined_ in anonymous namespace in the `.cpp`.

Placing them in anonymous namespace makes your result `.obj` smaller, since the compiler won't include their symbols in the output symboltable (cuz they're implementation, right? no one other than that one `.cpp` should know/be-able-to-access those functions/classes).

We should go over this to make sure you understand the differences.